### PR TITLE
Automated chart bump gatekeeper-0.5.1

### PR DESCRIPTION
**Pull request by d2iq-dispatch**
**What type of PR is this?**
Chore

**What this PR does/ why we need it**:
upgrade gatekeeper addon to newer version

**Which issue(s) this PR fixes**:
https://jira.d2iq.com/browse/D2IQ-73506

**Special notes for your reviewer**:
The upstream chart is located at https://github.com/open-policy-agent/gatekeeper/tree/master/charts/gatekeeper

I have done the manual upgrade test. Here are the steps that I followed:
```
1. Provisioned a Konvoy 1.6 cluster with gatekeeper enabled
2. Created a gatekeeper policy to validate allowed repository that a pod can use
     ( Here is the configuration that I used to test https://github.com/arbhoj/gatekeeper-policies/tree/master/allowed-repos-deployment-example)
3. Validated that the policy is working 
4. Upgraded gatekeeper using this chart
5. Validated that the policy is still working
```

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Review by s12chung — APPROVED** [review]

File: addons/gatekeeper/gatekeeper.yaml [hunk 1]
```diff
@@ -34,15 +33,10 @@ spec:
...
+      replicas: 2
       image:
-        tag: v3.0.4-beta.1
-      resources:
-        limits:
-          cpu: 500m
-          memory: 500Mi
-        requests:
-          cpu: 200m
-          memory: 300Mi
```
Comment (joejulian) [inline addons/gatekeeper/gatekeeper.yaml:48]: Unless these are the new chart defaults, we include these as our opinionated choices.

**Review by joejulian — CHANGES_REQUESTED** [review]

File: addons/gatekeeper/gatekeeper.yaml (hunk 1 above)
Comment (GoelDeepak) [reply addons/gatekeeper/gatekeeper.yaml:48]: There are new chart defaults https://github.com/mesosphere/charts/blob/master/staging/gatekeeper/values.yaml#L56. I am open-minded if we want to keep these vs chart defaults. I don't have enough data point about how we arrived at those values vs upstream. In the absence of that data point, I was favoring upstream.

Comment (alejandroEsc) [reply addons/gatekeeper/gatekeeper.yaml:48]: these were probably arrived at via MWT. I would suggest we keep these values for now and dial them up/down based on the next mwt run.

Comment (hectorj2f) [reply addons/gatekeeper/gatekeeper.yaml:48]: The values were extracted by our learnings to keep gatekeeper running healthy while doing load tests.

Comment (hectorj2f) [reply addons/gatekeeper/gatekeeper.yaml:48]: It seems the code of gatekeeper changed now. We are not going to exercise it during this new MWT. So we can keep the defaulted ones in the chart. The gatekeeper initial resource allocation were lower before, so we experienced OOM issues during our MWT tests.

Comment (GoelDeepak) [reply addons/gatekeeper/gatekeeper.yaml:48]: Sounds good.

**Review by joejulian — APPROVED** [review]

**Pull request merged**